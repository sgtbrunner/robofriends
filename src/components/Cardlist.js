import React from 'react';
import Card from './Card';

const Cardlist = ({ robots }) => {
  return (
    <div>
      {
			    robots.map((user) => {
			      return (
  <Card
    name={user.name}
    email={user.email}
    key={user.id}
  />
			      );
			    })
			}
    </div>
  );
};

export default Cardlist;
