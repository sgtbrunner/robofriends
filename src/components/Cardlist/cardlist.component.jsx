import React from 'react';
import PropTypes from 'prop-types';

import Card from '../Card';

const Cardlist = ({ robots }) => (
  <div>
    {robots.map((user) => {
      const { name, email, id } = user;
      return <Card name={name} email={email} id={id} key={id} />;
    })}
  </div>
);

Cardlist.propTypes = {
  robots: PropTypes.array.isRequired,
};

export default Cardlist;
