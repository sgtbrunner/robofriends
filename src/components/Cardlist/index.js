import Cardlist from './cardlist.component';

export default Cardlist;
