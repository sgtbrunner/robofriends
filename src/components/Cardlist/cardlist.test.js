import React from 'react';
import { shallow } from 'enzyme';
import Cardlist from './cardlist.component';

it('expect to render Card component', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'John Locke',
      username: 'locke',
      email: 'locke@gmail.com',
    },
  ];
  expect(shallow(<Cardlist robots={mockRobots} />).debug()).toMatchSnapshot();
});
