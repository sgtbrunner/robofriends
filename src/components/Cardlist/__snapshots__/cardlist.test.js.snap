// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
"<div>
  <Card name=\\"John Locke\\" email=\\"locke@gmail.com\\" id={1} />
</div>"
`;
